docs/design/2020-06-24-placement-rules-in-sql.md

```diff
@@ -0,0 +1,366 @@
+* label
+* role
+* count
+
+The system table is a virtual table, which doesn’t store data. When querying the table, TiDB queries PD and integrates the result in a table format. That also means the metadata is stored on PD instead of TiKV.
+
+Advantages of building system table include:
+
+* It’s easy for users to filter and aggregate
+* There’s no need to support new grammar, and it’s easier to implement
+
+### Show placement
+
+But there’re a problem here. The system table only contains stored placement rules, and users cannot query the effective rule of one object from it.
+
+For example, table `t` has two partitions `p0` and `p1`, and a placement rule is added on `t`. If the user wants to query the working rule of `p0`. Then he’ll find no placement rule is defined for `p0` through the system table. Based on the rule priorities described before, he must query the placement rule on `t`. This procedure may be annoying.
+
+To simplify this procedure, a SHOW statement should be provided to query the effective rule for one specified object.
+
+The statement is in such a format:
+
+```sql
+SHOW PLACEMENT FOR TABLE {table_name | partition_name};
+SHOW PLACEMENT FOR INDEX index_name FROM {table_name | partition_name};
+```
+
+TiDB will automatically find the effective rule based on the rule priorities.
+
+The output of this statement contains these fields:
+
+* target: The object queried. It can be a table, partition, or index. For table and partition, it is shown in the format `TABLE table_name`; For index, it is shown in the format `INDEX index_name FROM table_name`.
+* placement: The original ALTER statement that affects the placement of this rule. Note that the object in this ALTER statement may be different with `target`. In the example above, if `target` is `p0`, the ALTER statement shown may be `ALTER TABLE t …`.
+
+### Show create table
+
+It’s useful to show rules in `SHOW CREATE TABLE` statement, because users can check the rules easily.
+
+Since data in TiDB can be imported to MySQL, the placement rules definition must be shown as a MySQL-compatible comment in such a format `/*T![placement] placement_gammar*/`, where `placement_gammar` can be recognized by TiDB. That means TiDB needs to support two approaches to define placement rules, one in `CREATE TABLE` and another in `ALTER`.
+
+This is complicated, and `ALTER` is able to satisfy most of the cases, so `SHOW CREATE TABLE` can be kept untouched so far.
+
+## Privilege management
+
+Privilege management is the same as before:
+
+* Executing `ALTER` statement requires `Alter` privilege
+* Querying `information_schema.placement_rules` only returns those placement rules on the tables that visible to the current user
+* `ADMIN SHOW DDL` statement can only be executed by those users who have `Super` privilege.
+
+## Ecosystem tools
+
+There’re many tools that are based on binlog or metadata, such as DM, Lighting, Syncer, etc. Placement rules need to be compatible with these tools.
```

Comment:
Lightning does not rely on binlog, not even the binlog position.

Reply:
Placement rules can also rely on metadata to be exported to the downstream, just like schema information. And Lightning relies on metadata.